scipy/stats/mstats_basic.py

```diff
@@ -622,6 +622,8 @@ def kendalltau(x, y, use_ties=True, use_missing=False, method='auto'):
             prob = 2.0/np.math.factorial(n)
         elif c == 1:
             prob = 2.0/np.math.factorial(n-1)
+        elif 2*c == (n*(n-1))//2:
+            pvalue = 1.0
```

Comment:
This should be `prob = 1.0`.  Test case:

```
from scipy.stats import mstats

x = [9, 2, 5, 6]
y = [4, 7, 9, 11]
result = mstats.kendalltau(x, y, method='exact')
```
With scipy master branch (version '1.5.0.dev0+fac296a'), we currently get
```
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
<ipython-input-8-0ed9ea83fd63> in <module>
----> 1 mstats.kendalltau(x, y, method='exact')

~/mc37scipy/lib/python3.7/site-packages/scipy-1.5.0.dev0+fac296a-py3.7-macosx-10.7-x86_64.egg/scipy/stats/mstats_basic.py in kendalltau(x, y, use_ties, use_missing, method)
    666         raise ValueError("Unknown method "+str(method)+" specified, please use auto, exact or asymptotic.")
    667 
--> 668     return KendalltauResult(tau, prob)
    669 
    670 
UnboundLocalError: local variable 'prob' referenced before assignment
```